legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalIngestorAbstract.java

```diff
@@ -324,14 +342,67 @@ public List<DatasetFilter> getLatestStagingFilters(RelationalConnection connecti
 
     // ---------- UTILITY METHODS ----------
 
-    private void evolveSchema()
+    private void validateDatasetsInitialization()
     {
+        // Validation: init(Connection) must have been invoked
+        if (this.executor == null)
+        {
+            throw new IllegalStateException("Executor not initialized, call init(Connection) before invoking this method!");
+        }
+        // Validation: init(Datasets) must have been invoked
+        if (!this.datasetsInitialized)
+        {
+            throw new IllegalStateException("Datasets not initialized, call init(Datasets) before invoking this method!");
+        }
+    }
```

Comment:
Can change the method names in the comments and the error messages to be more accurate: initExecutor and initDatasets respectively

Reply:
Sure, will change